### PR TITLE
Payments: multi-network support

### PR DESCRIPTION
**Pull request by Wiezzel**
Added `platform` field to `PaymentDetails` and `SchedulePayment`. Supporting multiple networks invalidates the assumption that every driver supports a single payment platform. Therefore, this additional information is needed to be able to make appropriate payment.

Implemented validation of payment platforms in PaymentProcessor to secure against cross-network replay attacks (i.e. making payment on testnet and claiming it to be a mainnet payment).

### CODE REVIEW

File: core/payment/src/cli.rs [hunk 1]
```diff
@@ -17,27 +17,39 @@ use crate::{wallet, DEFAULT_PAYMENT_DRIVER};
...
+    //     #[structopt(long)]
+    //     account: Option<String>,
+    //     #[structopt(long, default_value = DEFAULT_PAYMENT_DRIVER)]
+    //     driver: String,
+    //     #[structopt(long)]
+    //     network: Option<String>,
+    //     #[structopt(long)]
+    //     amount: String,
+    // },
+    Exit {
```
Comment (tworec) [inline core/payment/src/cli.rs:31]: pls add `///` comments to fill `--help` for all commands

File: core/payment/src/cli.rs [hunk 2]
```diff
@@ -56,17 +68,19 @@ pub enum PaymentCli {
         #[structopt(subcommand)]
```
Comment (tworec) [inline core/payment/src/cli.rs:68]: I'm not sure, but seems that it is the only `address` within payment CLI. Shouldn't it be renamed to `account` to unify CLI?

Comment (Wiezzel) [reply core/payment/src/cli.rs:68]: This is node ID, not payment account address. I'm not sure what to do with it. I don't even know if anyone is using this command.

File: core/payment/src/cli.rs (hunk 1 above)
Comment (Wiezzel) [reply core/payment/src/cli.rs:31]: Good idea. Will do if there's enough time.

**Review by maaktweluit — APPROVED** [review]
LGTM!

Tested all CLI Commands and invoice_flow scenario flavours

**Review by tworec — APPROVED** [review]

**Pull request merged**